src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAttributeProvider.java

```diff
@@ -104,7 +104,9 @@ protected List<UserAttributes> extractUserAttributes(
 
         // override from account
         map.put(OpenIdAttributesSet.NAME, name);
-        map.put(OpenIdAttributesSet.FAMILY_NAME, surname);
+        if (surname != null) {
```

Comment:
check with StringUtils.hasText() to also evaluate if it's empty or blank in addition to null